src/common/libflux/message.c

```diff
@@ -1727,47 +1751,43 @@ int flux_msg_sendzsock (void *sock, const flux_msg_t *msg)
     return flux_msg_sendzsock_ex (sock, msg, false);
 }
 
-static int recvzsock_frame (void *handle, zmsg_t *zmsg)
-{
-    zframe_t *zf = NULL;
-    zmq_msg_t zmqmsg;
-    int rv = -1;
-
-    zmq_msg_init (&zmqmsg);
-    if (zmq_recvmsg (handle, &zmqmsg, 0) < 0)
-        goto error;
-    if (!(zf = zframe_new (zmq_msg_data (&zmqmsg), zmq_msg_size (&zmqmsg))))
-        goto error;
-    zframe_set_more (zf, zsock_rcvmore (handle));
-    if (zmsg_append (zmsg, &zf))
-        goto error;
-    rv = 0;
-error:
-    if (rv < 0)
-        zframe_destroy (&zf);
-    zmq_msg_close (&zmqmsg);
-    return rv;
-}
-
 flux_msg_t *flux_msg_recvzsock (void *sock)
 {
     void *handle;
-    zmsg_t *zmsg = NULL;
+    struct msg_iovec *iov = NULL;
+    int iovcnt = 0;
     flux_msg_t *msg;
     flux_msg_t *rv = NULL;
 
     if (!sock) {
         errno = EINVAL;
         return NULL;
     }
-    if (!(zmsg = zmsg_new ())) {
-        errno = ENOMEM;
-        return NULL;
-    }
+
+    /* N.B. we need to store a zmq_msg_t for each iovec entry so that
+     * the memory is available during the call to iovec_to_msg().  We
+     * use the msg_iovec's "transport_data" field to store the entry
+     * and then clear/free it later.
+     */
     handle = zsock_resolve (sock);
     while (true) {
-        if (recvzsock_frame (handle, zmsg) < 0)
+        struct msg_iovec *tmp;
+        zmq_msg_t *transport_data;
```

Comment:
Shorter name for local variable?   Maybe `zmsg`?

Reply:
hehe, I intentionally tried to avoid `zmsg` b/c it could imply `zmsg_t`, which we just got rid of.  perhaps just `msgdata` or something like that.